Tệp số liệu ch4bt2.txt cho 40 quan sát về các biến BEER- mức chi tiêu cho bia/người/năm; Y - thu nhập/người/năm ($); AGE - tuổi; SEX- giới tính (1 nam; 0 nữ); học vấn:  D1 = 1 nếu tốt nghiệp trung học; D1= 0 đối với các trường hợp khác; D2 = 1 nếu tốt nghiệp cao đẳng, D2 = 0 đối với các trường hợp khác; D3 = 1 nếu tốt nghiệp đại học, D3 = 0 đối với các trường hợp khác.
      Hãy tính toán và lập luận đầy đủ để trả lời các câu hỏi sau:


In [108]:
library(lmtest)
library(car)
library(carData)
library(hexView)
library(strucchange)
library(fBasics)

In [109]:
df <- readEViews("ch4bt2.wf1")
df

Warning message in readEViews("ch4bt2.wf1"):
"Skipping boilerplate variable
"
Warning message in readEViews("ch4bt2.wf1"):
"Skipping boilerplate variable
"


Date       AGE BE  D1 D2 D3 SEX Y     
1  0001-01-01 25  109 0  0  0  1    15000
2  0002-01-01 45    0 0  0  0  1    30000
3  0003-01-01 20    0 0  0  0  1    12000
4  0004-01-01 28  108 0  0  0  1    20000
5  0005-01-01 25  220 1  0  0  1    15000
6  0006-01-01 35  189 1  0  0  1    30000
7  0007-01-01 40   64 1  0  0  1    12000
8  0008-01-01 22  262 1  0  0  1    12000
9  0009-01-01 30   64 1  0  0  1    28000
10 0010-01-01 21   35 1  0  0  1    22000
11 0011-01-01 40   94 1  0  0  1    44000
12 0012-01-01 21   71 0  1  0  1    10000
13 0013-01-01 45  403 0  1  0  1   222000
14 0014-01-01 36   41 0  1  0  1    32000
15 0015-01-01 36   10 0  1  0  1    45000
16 0016-01-01 40  110 0  1  0  1    55000
17 0017-01-01 23  239 0  1  0  1    29000
18 0018-01-01 32   63 0  1  0  1    39000
19 0019-01-01 52    0 0  1  0  1    70000
20 0020-01-01 30  106 0  0  1  1    55000
21 0021-01-01 45    0 0  0  1  0    90000
22 0022-01-01 32  141 0  0  0  0     6000
23 0023-01-01 20  299 0  0  0  0    18000
24 0024-01-01 55  148 0  0  0  0    55000
25 0025-01-01 18  424 1  0  0  0    10000
26 0026-01-01 30  242 1  0  0  0    23000
27 0027-01-01 45  119 1  0  0  0    35000
28 0028-01-01 40  338 1  0  0  0    38000
29 0029-01-01 50  135 1  0  0  0    45000
30 0030-01-01 32  590 1  0  0  0    85000
31 0031-01-01 30  324 1  0  0  0    22000
32 0032-01-01 51   87 1  0  0  0    25000
33 0033-01-01 22  395 0  1  0  0    29000
34 0034-01-01 40  513 0  1  0  0   132000
35 0035-01-01 30   56 0  1  0  0    35000
36 0036-01-01 36  400 0  1  0  0    80000
37 0037-01-01 27  384 0  1  0  0    55000
38 0038-01-01 24  262 0  1  0  0    30000
39 0039-01-01 21  336 0  1  0  0    27000
40 0040-01-01 45  281 0  0  1  0    80000

## Câu 1

Hãy ước lượng hàm tiêu dùng về bia phụ thuộc vào thu nhập, tuổi, giới tính, học vấn.

In [110]:
hq <- lm(data = df,BE~Y+AGE+SEX+D1+D2+D3)
summary(hq)


Call:
lm(formula = BE ~ Y + AGE + SEX + D1 + D2 + D3, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-205.68  -43.40   17.55   57.09  126.35 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.373e+02  6.212e+01   7.040 4.69e-08 ***
Y            2.811e-03  4.712e-04   5.966 1.07e-06 ***
AGE         -8.967e+00  1.643e+00  -5.457 4.79e-06 ***
SEX         -1.693e+02  2.843e+01  -5.955 1.10e-06 ***
D1           7.485e+01  4.085e+01   1.832   0.0759 .  
D2          -5.014e+00  4.416e+01  -0.114   0.9103    
D3          -1.040e+02  6.510e+01  -1.598   0.1196    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 88.76 on 33 degrees of freedom
Multiple R-squared:  0.7256,	Adjusted R-squared:  0.6757 
F-statistic: 14.55 on 6 and 33 DF,  p-value: 4.808e-08


## Câu 2

Sử dụng kiểm đình F để xem có bỏ được D2 và D3 ra khỏi mô hình hay không? 

In [111]:
hq0 <- lm(data=df,BE~Y+AGE+SEX+D1)
rur <- summary(hq)$r.squared
rr <- summary(hq0)$r.squared

In [112]:
fqs <- ((rur-rr)/2)/((1-rur)/33)
fqs

[1] 1.597317

In [113]:
qf(0.95, df1 = 2, df2 = 33)

[1] 3.284918

Vậy có thể bỏ D2 và D3 ra khỏi mô hình

##### Cách 2

In [114]:
anova(hq0,hq)

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 35     285179.6 NA       NA        NA        NA
2 33     260008.9  2 25170.71  1.597317 0.2176961

##### Cách 3

In [115]:
linearHypothesis(hq,c("D2=0","D3=0"))

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 35     285179.6 NA       NA        NA        NA
2 33     260008.9  2 25170.71  1.597317 0.2176961

Với mức ý nghĩa 5%, có thể bỏ D2 và D3 ra khỏi mô hình

## Câu 3

Ước lương lại mô hình và kiểm định xem dạng hàm có phù hợp không?

In [116]:
resettest(hq, power = c(2,3))


	RESET test

data:  hq
RESET = 2.97, df1 = 2, df2 = 31, p-value = 0.06605


Với mức ý nghĩa 5%, dạng hàm là phù hợp

## Câu 4

Phương sai của sai số có đồng đều không?

In [117]:
df$phandu <- resid(hq)
summary(lm(data = df, phandu ~ Y+AGE+I(Y^2)+I(AGE^2)+I(Y*AGE)))


Call:
lm(formula = phandu ~ Y + AGE + I(Y^2) + I(AGE^2) + I(Y * AGE), 
    data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-214.41  -51.19   16.25   55.02  129.83 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)
(Intercept) -2.096e+01  1.650e+02  -0.127    0.900
Y            4.334e-03  3.433e-03   1.263    0.215
AGE         -3.023e+00  1.117e+01  -0.271    0.788
I(Y^2)      -1.840e-09  7.069e-09  -0.260    0.796
I(AGE^2)     7.711e-02  1.811e-01   0.426    0.673
I(Y * AGE)  -9.379e-05  1.012e-04  -0.927    0.360

Residual standard error: 84.1 on 34 degrees of freedom
Multiple R-squared:  0.07511,	Adjusted R-squared:  -0.06091 
F-statistic: 0.5522 on 5 and 34 DF,  p-value: 0.7355


In [118]:
bptest(hq, ~ I(Y^2)+I(AGE^2)+I(Y*AGE), data = df)


	studentized Breusch-Pagan test

data:  hq
BP = 4.0415, df = 3, p-value = 0.257


Phương sai sai số đồng đều

## Câu 5

u có phân bố chuẩn? với mức ý nghãi 5%.

In [119]:
jarqueberaTest(hq$residuals)


Title:
 Jarque - Bera Normalality Test

Test Results:
  STATISTIC:
    X-squared: 4.2341
  P VALUE:
    Asymptotic p Value: 0.1204 

Description:
 Sun Mar 12 19:49:25 2023 by user: Admin


Với mức ý nghĩa 5%, u có phân phối chuẩn

## Câu 6

Với các yếu khác như nhau, nam và nữ có tiêu dùng bia khác nhau không?  Vì sao. Mức khác nhau là bao nhiêu?

In [120]:
hq1 <- lm(data = df,BE~SEX)
summary(hq1)


Call:
lm(formula = BE ~ SEX, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-273.70 -101.90   -7.55   87.27  316.30 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   273.70      29.86   9.166 3.63e-11 ***
SEX          -164.30      42.23  -3.891  0.00039 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 133.5 on 38 degrees of freedom
Multiple R-squared:  0.2849,	Adjusted R-squared:  0.266 
F-statistic: 15.14 on 1 and 38 DF,  p-value: 0.0003902


Với các yếu khác như nhau, nam và nữ có tiêu dùng bia là khác nhau 

Vì sao hệ số của biến SEX có ý nghĩa. 

Nam chi tiêu cho bia trung bình ít hơn nữ -164.30 đơn vị

## Câu 7 

Một nữ, 28 tuổi, chưa tốt nghiệp trung học, có thu nhập trung bình/năm bằng 25000 \$/năm và một nam 30 tuổi, tốt nghiệp trung học, có thu nhập 27000\$/năm. Hãy dự báo khoảng tiêu dùng beer trung bình trong năm với hệ số tin cậy 95% của hai người này

In [121]:
Y <- c(25000,27000)
AGE <- c(28,30)
SEX <- c(0,1)
D1 <- c(0,1)
D2 <- c(0,0)
D3 <- c(0,0)
predic <- predict(hq,data.frame(Y,AGE,SEX,D1,D2,D3),interval="confidence",level=0.95)
predic

fit      lwr       upr     
1 256.5161 178.77724 334.2551
2 149.7629  93.09669 206.4291